crates/noirc_evaluator/src/ssa/acir_gen/operations/store.rs

```diff
@@ -1,34 +1,66 @@
+use acvm::FieldElement;
+
 use crate::{
     ssa::{
-        acir_gen::{internal_var_cache::InternalVarCache, memory_map::MemoryMap, InternalVar},
+        acir_gen::{
+            constraints::{add, mul_with_witness, subtract},
+            internal_var_cache::InternalVarCache,
+            memory_map::MemoryMap,
+            InternalVar,
+        },
         context::SsaContext,
-        mem::{self, ArrayId},
-        node::NodeId,
+        mem::{self},
+        node::Operation,
     },
     Evaluator,
 };
 
 pub(crate) fn evaluate(
-    array_id: ArrayId,
-    index: NodeId,
-    value: NodeId,
+    store: &Operation,
     memory_map: &mut MemoryMap,
     var_cache: &mut InternalVarCache,
     evaluator: &mut Evaluator,
     ctx: &SsaContext,
 ) -> Option<InternalVar> {
-    //maps the address to the rhs if address is known at compile time
-    let index = var_cache.get_or_compute_internal_var_unwrap(index, evaluator, ctx);
-    let value = var_cache.get_or_compute_internal_var_unwrap(value, evaluator, ctx);
+    if let Operation::Store { array_id, index, value, predicate } = *store {
```

Comment:
Looks like this was changed to take the whole Operation for convenience of not passing multiple parameters. Rather than passing it and unwrapping here, perhaps we should name the struct inside `Operation::Store` so we can pass the variant as one value and not need to unwrap it. So the enum definition would look like `Operation::Store(Store),`

Reply:
It was not for convenience, clippy was complaining that it was too many arguments

Reply:
Ah, I've run into that as well. Making the struct is something we can do but it can be done in another PR and I will not block this one if you do not want to. Feel free to resolve this comment.

Reply:
Thank you, I agree that we should use a struct inside the enum but I prefer that it is done in another PR.